plugins/normalize-whitespace/prism-normalize-whitespace.js

```diff
@@ -161,6 +173,27 @@
 			return;
 		}
 
+		if (env.settings == null) { env.settings = {}; }
+
+		// Read settings from 'data-' attributes
+		for (var key in settingsConfig) {
+			if (Object.hasOwnProperty.call(settingsConfig, key)) {
+				var settingType = settingsConfig[key];
+				if (pre.hasAttribute('data-' + key)) {
+					var value = pre.getAttribute('data-' + key);
+					switch (settingType) {
+						case 'boolean':
+							value = (value === 'true');
+							break;
+						case 'number':
+							value = (isNaN(+value) ? 0 : +value);
+							break;
+					}
```

Comment:
This is too brittle. Why is the value initialized to the unparsed value? Why are invalid values just silently accepted and parsed using some kind of default value? Why is there no default case?

Anything that isn't a valid boolean should be ignored. Similarly, anything that isn't a valid number should be ignored (0 as a default doesn't even make sense for a lot of settings).

---

Also, users should be able to input `true` like this `<pre data-prop>`.

Reply:
I can add the `<pre data-prop>` use case. But I would do any kind of validation in the actual functions that are called. Otherwise you're validating the data-props but not when calling the functions vis JS. I was concerning this PR with adding the ability to pass props without using JS.

The reason for setting the number to 0 if invalid is all the functions that take a number in this plugin default to 0 if something is null/undefined. But I can change this to just ignore NaN (functionally it would be the same thing)

Reply:
There is no default case as its impossible to not be a `boolean` or a `number`. This data is hard-coded on line 36.

Reply:
> The reason for setting the number to 0 if invalid is all the functions that take a number in this plugin default to 0 if something is null/undefined.

Their default values might change and should not affect argument parsing.

Reply:
Yes I corrected this